### PR TITLE
feat(verify): Pass `service` param when verifying email address.

### PR DESCRIPTION
**Pull request by rfk**
A work-in-progress to pass the `service` param through when verifying an email code.  This would allow us to tailor the content of the post-verification email based on what service is being accessed.  Connects to https://github.com/mozilla/fxa-auth-server/issues/1249

Not ready yet as it obviously needs tests, but @shane-tomlinson does this seem a reasonable approach to you?

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 58bed41a1b13ae678d2587a956acab15dcc31b11
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by shane-tomlinson** [comment]
@rfk - seems totally reasonable.

File: app/scripts/models/account.js [hunk 1]
```diff
@@ -404,11 +404,12 @@ define(function (require, exports, module) {
      * @param {string} code - the verification code
      * @returns {promise} - resolves when complete
```
Comment (vladikoff) [inline app/scripts/models/account.js]: needs doc update (`options` needs to be added)

**Comment by shane-tomlinson** [comment]
@rfk - do you want this shipped with train-63? If so, can you add the shipit label?

**Comment by rfk** [comment]
Hmmm...there's a deployment consideration here.  IIUC we have to deploy https://github.com/mozilla/fxa-auth-server/pull/1268 to production in order to support the new field in the request body, before we can deploy this change in content-server.

Anyways, @vbudhram per our discussion today and assuming you're still happy to take it on, I'm going to hand this off to you to continue work on.

**Comment by shane-tomlinson** [comment]
> Hmmm...there's a deployment consideration here. IIUC we have to deploy mozilla/fxa-auth-server#1268 to production in order to support the new field in the request body, before we can deploy this change in content-server.

You may be able to deploy the content server code without the corresponding JS client code which passes the option to the auth server.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 58bed41a1b13ae678d2587a956acab15dcc31b11
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

 * Commit: 984c5836aab87b14b4ecd98c4a8b7155dcab39cd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

 * Commit: 892484825a20feb4295216e5a713a99ed8fee32c
  * Your commit message body contains a line that is longer than 100 characters
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

 * Commit: d45a60cc79aad37af1a64eadd07b695bcc29078c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

 * Commit: 80d57c9b7d4dfb19fcbd3a9f173faa410d17488b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 13852136ec13812712fbb20f09b4ce2c2e5820d3
  * Your subject line is longer than 100 characters

 * Commit: 63594b38aa0630c041145a28716f9b631867ef7a
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 58bed41a1b13ae678d2587a956acab15dcc31b11
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by vbudhram** [comment]
@shane-tomlinson I think there are more test cases that could be added but am drawing a blank on them at the moment. If not, mind a r?

**Comment by shane-tomlinson** [comment]
ref https://github.com/mozilla/fxa-js-client/pull/199

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson I think there are more test cases that could be added but am drawing a blank on them at the moment. If not, mind a r?

I added a bunch of functional tests to ensure the emails arrive, or didn't, depending on which flow the user is in. One unexpected outcome, sign up via accounts.firefox.com directly sends a post-verification email.

@vbudhram - since I added a bunch of new test code, mind doing an r on the new tests?

File: tests/lib/restmail.js [hunk 2]
```diff
@@ -7,15 +7,33 @@ define([
...
+  /**
+   * Wait for an email.
+   *
+   * @param {string} uri - email uri
+   * @param {number} number
+   * @param {object} [options]
+   *   @param {number} [options.minAttemptsBeforeLog] - Minimum number of
+   *   attempts before attempts are logged. Defaults to 2.
+   *   @param {number} [options.maxAttempts] - number of email fetch attempts
+   *   to make. Defaults to Infinity.
```
Comment (vbudhram) [inline tests/lib/restmail.js]: Was there a reason to default to infinity? Maybe reuse 10 maxAttempts from above?

**Comment by vbudhram** [comment]
> One unexpected outcome, sign up via accounts.firefox.com directly sends a post-verification email.

I believe that is the intended response. If the `service` is not specified, the [auth server](https://github.com/mozilla/fxa-auth-server/blob/master/lib/routes/account.js#L1082) will fallback to sending the email.

**Comment by vbudhram** [comment]
LGTM 👍, pending tests, r+

**Comment by rfk** [comment]
>  > One unexpected outcome, sign up via accounts.firefox.com directly sends a post-verification email.
> I believe that is the intended response. If the service is not specified, the auth server will fallback
>  to sending the email.

Right.  When we get a more generic post-verification email, we can send that one by default and keep th current one just for sync.

**Comment by rfk** [comment]
It's not obvious to me whether the CircleCI failures here are test bustage from this patch, or unrelated instability.  @vbudhram what's the next action on this bug?  I'd like us to ship it in train-64 if it all possible since it continues to be a source of weirdness for users signing up in AMO and TestPilot.

**Comment by vbudhram** [comment]
After rebase, looks good to go.

**Pull request merged**